### PR TITLE
[#10920] Add front - end unit tests to Comment edit, Comment row and Adding question panel component

### PR DESCRIPTION
**Pull request by SimonHasak**
Part of #10920

In environment.ts fixed import from 'import { config } from './config';' to 'import { config } from './config.template';'. I noticed that with npm run coverage why so many tests fails...

Comment edit component unit tests added.
Comment row component unit tests added.
Adding question panel component  component unit tests added.

I tried to cover what I saw was missing.

### CODE REVIEW

**Comment by SimonHasak** [comment]
@t-cheepeng please can you take a look please? I wasn`t able to add reviewers that`s why I am tagging in the comment. We have a school subject which obligate to contribute to open source with some tests by the way :D.

File: angular.json [hunk 1]
```diff
@@ -115,5 +115,8 @@
       }
     }
   },
-  "defaultProject": "teammates"
+  "defaultProject": "teammates",
+  "cli": {
```
Comment (t-cheepeng) [inline angular.json:119]: This is probably unintended. Please remove

File: src/web/environments/environment.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { config } from './config';
+import { config } from './config.template';
```
Comment (t-cheepeng) [inline src/web/environments/environment.ts:1]: same here

**Review by t-cheepeng — CHANGES_REQUESTED** [review]

File: angular.json (hunk 1 above)
Comment (SimonHasak) [reply angular.json]: Fixed.

File: src/web/environments/environment.ts (hunk 2 above)
Comment (SimonHasak) [reply src/web/environments/environment.ts:1]: The link was broken and tests were not passing so I updated the link path and its fine now. But sure I will remove it if you will persist.

Comment (wkurniawan07) [reply src/web/environments/environment.ts:1]: @SimonHasak template files are not meant to be used directly. If you have followed the step mentioned in https://github.com/TEAMMATES/teammates/blob/master/docs/setting-up.md#step-3-set-up-project-specific-settings-and-dependencies, you should have a `config.ts` file.

Comment (SimonHasak) [reply src/web/environments/environment.ts]: Yes I see now. Thanks for explanation ;).

**Review by wkurniawan07 — CHANGES_REQUESTED** [review]

**Review by wkurniawan07 — CHANGES_REQUESTED** [review]

File: src/web/app/components/adding-question-panel/adding-question-panel.component.spec.ts [hunk 3]
```diff
@@ -36,4 +36,42 @@ describe('AddingQuestionPanelComponent', () => {
...
+
+    spyOn(component.copyQuestionsFromOtherSessionsEvent, 'emit');
+    button.click();
+    fixture.detectChanges();
+
+    component.copyQuestionsFromOtherSessionsHandler();
+
+    expect(component.copyQuestionsFromOtherSessionsEvent.emit).toHaveBeenCalled();
+  });
 });
```
Comment (daongochieu2810) [inline src/web/app/components/adding-question-panel/adding-question-panel.component.spec.ts:77]: these 3 tests seem very similar to each other, maybe we can use a for loop and compress them into 1 single test

File: src/web/app/components/comment-box/comment-edit-form/comment-edit-form.component.spec.ts [hunk 4]
```diff
@@ -37,10 +63,136 @@ describe('CommentEditFormComponent', () => {
...
+
+    spyOn(component.modelChange, 'emit');
+
+    button.click();
+    fixture.detectChanges();
+
+    component.triggerModelChangeBatch(object);
+
+    expect(component.modelChange.emit).toHaveBeenCalledWith({ ...commentModel, ...object });
+  });
```
Comment (daongochieu2810) [inline src/web/app/components/comment-box/comment-edit-form/comment-edit-form.component.spec.ts:126]: Same as above for these 3

File: src/web/app/components/comment-box/comment-edit-form/comment-edit-form.component.spec.ts [hunk 5]
```diff
@@ -37,10 +63,136 @@ describe('CommentEditFormComponent', () => {
...
+
+  it('should disallowToSee visibility in modifyVisibilityControl method', () => {
+    const spy1: any = spyOn(component.visibilityStateMachine, 'disallowToSee');
+    const spy2: any = spyOn(component, 'triggerModelChangeBatch');
+
+    component.modifyVisibilityControl(false, visibilityType, visibilityControl);
+
+    expect(spy1).toHaveBeenCalledWith(visibilityType, visibilityControl);
+    expect(spy2).toHaveBeenCalledWith(modelChangeBatch);
+  });
```
Comment (daongochieu2810) [inline src/web/app/components/comment-box/comment-edit-form/comment-edit-form.component.spec.ts:146]: we can combine these into 1 test

File: src/web/app/components/comment-box/comment-edit-form/comment-edit-form.component.spec.ts [hunk 6]
```diff
@@ -37,10 +63,136 @@ describe('CommentEditFormComponent', () => {
...
```
Comment (daongochieu2810) [inline src/web/app/components/comment-box/comment-edit-form/comment-edit-form.component.spec.ts:156]: this seems like the same test as the one right above?

**Comment by moziliar** [comment]
@SimonHasak ping if you have time to respond to the reviews.

**Comment by daongochieu2810** [comment]
@SimonHasak let us know if you have resolved all reviews, and failing CI/CD tests

**Comment by SimonHasak** [comment]
@daongochieu2810 I updated branch. Should be fine now.

File: package-lock.json [hunk 7]
```diff
@@ -10876,9 +10876,9 @@
       "integrity": "sha512-nnbWWOkoWyUsTjKrhgD0dcz22mdkSnpYqbEjIm2nhwhuxlSkpywJmBo8h0ZqJdkp73mb90SssHkN4rsRaBAfAA=="
     },
     "nan": {
-      "version": "2.14.1",
-      "resolved": "https://registry.npmjs.org/nan/-/nan-2.14.1.tgz",
-      "integrity": "sha512-isWHgVjnFjh2x2yuJ/tj3JbwoHu3UC2dX5G/88Cm24yB6YopVgxvBObDY7n5xW6ExmFhJpSEQqFPvq9zaXc8Jw==",
+      "version": "2.15.0",
+      "resolved": "https://registry.npmjs.org/nan/-/nan-2.15.0.tgz",
+      "integrity": "sha512-8ZtvEnA2c5aYCZYd1cvgdnU6cqwixRoYg70xPLWUws5ORTa/lnw+u4amixRS/Ac5U5mQVgp9pnlSUnbNWFaWZQ==",
```
Comment (daongochieu2810) [inline package-lock.json:10881]: I think these changes are not necessary?

**Review by daongochieu2810 — CHANGES_REQUESTED** [review]

**Review by daongochieu2810 — APPROVED** [review]

File: src/web/app/components/comment-box/comment-row/comment-row.component.spec.ts [hunk 8]
```diff
@@ -46,4 +51,55 @@ describe('CommentRowComponent', () => {
...
+      commentEditFormModel: {
+        commentText: '',
+        isUsingCustomVisibilities: false,
+        showCommentTo: [],
+        showGiverNameTo: [],
+      },
+      isEditing: false,
+    };
+
+    spyOn(component.modelChange, 'emit').and.returnValue(of({ ...commentRowModel, ...object }));
```
Comment (jianhandev) [inline src/web/app/components/comment-box/comment-row/comment-row.component.spec.ts:96]: @SimonHasak I think you forgot to change this part?

**Comment by daongochieu2810** [comment]
@SimonHasak Let us know if you are still working on this issue, thanks!

**Comment by hhdqirui** [comment]
Hi @SimonHasak, kind reminder on this issue.

**Comment by halfwhole** [comment]
Hey @SimonHasak, thanks for your contribution, but we'll be closing this PR due to a lack of activity these few months - however, do feel free to re-open it if you wish to continue working on it. Thanks!

**Pull request closed without merging**